File: presentation/src/main/java/io/forus/me/android/presentation/view/fragment/FragmentHelper.java
package io.forus.me.android.presentation.view.fragment;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import android.view.View;

import io.forus.me.android.presentation.navigation.Navigator;

public class FragmentHelper {

    public static void setHomeIconToolbar(final AppCompatActivity _activity, Toolbar toolbar, View profile, boolean allowBack){
       if (allowBack) {
            _activity.getSupportActionBar().setHomeButtonEnabled(true);
            _activity.getSupportActionBar().setDisplayHomeAsUpEnabled(true);
           profile.setVisibility(View.GONE);
        } else {
            _activity.getSupportActionBar().setHomeButtonEnabled(false);
            _activity.getSupportActionBar().setDisplayHomeAsUpEnabled(false);

           profile.setVisibility(View.VISIBLE);

           profile.setOnClickListener(new View.OnClickListener() {
               @Override
               public void onClick(View view) {
                   (new Navigator()).navigateToAccount(_activity);
               }
           });
        }
    }
}

File: xvgfdfg/presentation/src/main/java/io/forus/me/android/presentation/view/fragment/FragmentHelper.kt
package io.forus.me.android.presentation.view.fragment

import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import io.forus.me.android.presentation.navigation.Navigator

object FragmentHelper {
    fun setHomeIconToolbar(
        _activity: AppCompatActivity,
        toolbar: Toolbar?,
        profile: View?,
        allowBack: Boolean
    ) {
        if (allowBack) {
            _activity.supportActionBar?.setHomeButtonEnabled(true)
            _activity.supportActionBar?.setDisplayHomeAsUpEnabled(true)
            profile?.visibility = View.GONE
        } else {
            _activity.supportActionBar?.setHomeButtonEnabled(false)
            _activity.supportActionBar?.setDisplayHomeAsUpEnabled(false)
            profile?.visibility = View.VISIBLE
            profile?.setOnClickListener { Navigator().navigateToAccount(_activity) }
        }
    }
}